bsb/config/_parse_types.py

```diff
@@ -18,7 +18,7 @@ def _location_parts(self, carry):
         while (parent := parent._parent) is not None:
             if parent._parent is not None:
                 carry.insert(0, parent._key)
-        carry.append(self._key)
+        carry.append(self._key if self._key is not None else "")
```

Comment:
```suggestion
        carry.append(self._key or "")
```

is the same IF `self._key` is `str | None`, which I think it is